tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/RetryConditionTest.java

```diff
@@ -215,4 +227,144 @@ public void testClassLevelRetryWithAbortOnTrue() {
+            assertEquals("Simulated error", e.getMessage());
+        }
+        // 2 retries
+        assertEquals(2, asyncRetryClient.getCountInvocationsServBFailException());
+    }
+
+    /**
+     * Temporary error. Will retry 3 times, the first 2 will fail.
+     * ServiceC uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous}.
+     */
+    @Test
+    public void testRetrySuccess() {
+        final CompletionStage<String> future = asyncRetryClient.serviceC();
+
+        assertCompleteOk(future, "Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServC());
+    }
+
+    /**
+     * Temporary error. Will retry 3 times, the first 2 will fail deep in a CompletableFuture chained execution.
+     * ServiceD uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and chains 2 CompletableFutures.
+     */
+    @Test
+    public void testRetryChainSuccess() {
+        final CompletionStage<String> future = asyncRetryClient.serviceD();
+
+        assertCompleteOk(future, "Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServD());
+    }
+
+    /**
+     * Persistent Error condition. Will retry 3 times and still throw exception.
+     * ServiceE will always return IOException.
+     */
+    @Test
+    public void testRetryChainExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceE();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(3, asyncRetryClient.getCountInvocationsServE());
+    }
+
+    /**
+     * Persistent Error condition. Will retry 3 times and still throw exception.
+     * ServiceF will always return IOException.
+     */
+    @Test
+    public void testRetryParallelExceptionally() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceG();
+
+        assertCompleteExceptionally(future, IOException.class, "Simulated error");
+        assertEquals(3, asyncRetryClient.getCountInvocationsServG());
+    }
+
+    /**
+     * Temporary error. Will retry 3 times, the first 2 will fail in a CompletableFuture parallel execution.
+     * ServiceG uses {@link org.eclipse.microprofile.faulttolerance.Asynchronous} and 2 CompletableFutures.
+     */
+    @Test
+    public void testRetryParallelSuccess() {
+        CompletableFuture<String> future = new CompletableFuture<>();
+
+        asyncRetryClient.serviceG();
+
+        assertCompleteOk(future, "Success then Success");
+        assertEquals(2, asyncRetryClient.getCountInvocationsServG());
+    }
+
+    private void assertCompleteExceptionally(final CompletionStage<String> future,
+                                             final Class<? extends Throwable> exceptionClass,
+                                             final String exceptionMessage) {
+        try {
+            future.toCompletableFuture().get(200, TimeUnit.MILLISECONDS);
```

Comment:
`toCompletableFuture` is an optional method, we probably shouldn't rely on it.

There's already a utility in `CompletableFutureHelper.toCompletableFuture()` which creates a new completable future from a `CompletionStage` without using this method. (Remember to also add the class to the list in `deploy()`)